src/main/groovy/io/openliberty/tools/gradle/utils/LooseEarApplication.groovy

```diff
@@ -1,42 +1,50 @@
-import io.openliberty.tools.common.plugins.config.LooseApplication
+import org.w3c.dom.Element
 
 public class LooseEarApplication extends LooseApplication {
     
     protected Task task;
+    protected Logger logger;
 
-    public LooseEarApplication(Task task, LooseConfigData config) {
-        super(task.getProject().getBuildDir().getAbsolutePath(), config)
+    public LooseEarApplication(Task task, LooseConfigData config, Logger logger) {
+        super(task.getProject().getLayout().getBuildDirectory().getAsFile().get().getAbsolutePath(), config)
         this.task = task
+        this.logger = logger
     }
 
     public void addSourceDir() throws Exception {
-        File sourceDir = new File(task.getProject().path.replace(":","") + "/" + task.getProject().getConvention().getPlugin(EarPluginConvention.class).getAppDirName())
-        config.addDir(sourceDir, "/")
+        if (task.getProject().getPlugins().hasPlugin("ear")) {
+            Ear ear = (Ear) task.getProject().ear
+            File sourceDir = new File(task.getProject().path.replace(":","") + "/" + ear.getAppDirectory().getAsFile().get().getPath())
+            config.addDir(sourceDir, "/")
+        }
     }
 
     public void addApplicationXmlFile() throws Exception {
-        String applicationName = "/" + task.getProject().getConvention().getPlugin(EarPluginConvention.class).getDeploymentDescriptor().getFileName()
-        File applicationXmlFile = new File(task.getProject().path.replace(":","") + "/" + task.getProject().getConvention().getPlugin(EarPluginConvention.class).getAppDirName() + "/META-INF/" + applicationName)
-        if (applicationXmlFile.exists()) {
-            config.addFile(applicationXmlFile, "/META-INF/application.xml");
+        String applicationName = "/application.xml"
+        File applicationXmlFile;
+        if (task.getProject().getPlugins().hasPlugin("ear")) {
+            Ear ear = (Ear) task.getProject().ear
+            if (ear.getDeploymentDescriptor() != null) {
+                applicationName = "/" + ear.getDeploymentDescriptor().getFileName()
+            }
+            applicationXmlFile = new File(task.getProject().path.replace(":", "") + "/" + ear.getAppDirectory().getAsFile().get().getAbsolutePath() + "/META-INF/" + applicationName)
+            if (applicationXmlFile.exists()) {
+                config.addFile(applicationXmlFile, "/META-INF/application.xml");
+            }
         }
-        else {
+        if (applicationXmlFile == null || !applicationXmlFile.exists()) {
             applicationXmlFile = new File(task.getDestinationDirectory().get().getAsFile().getParentFile().getAbsolutePath() + "/tmp/ear" + applicationName);
             config.addFile(applicationXmlFile, "/META-INF/application.xml");
+            logger.warn("Could not get the application.xml file location from the EAR plugin because it is not configured. The file may not be added correctly to the application archive.")
```

Comment:
Since we had default behavior before without a warning, I think we should not actually log a warning here. I didn't notice the default behavior since you had removed it previously.